Add optional name to MigrationInterface (fixes #3933)

leonardofalk opened the pull request:
This fixes a problem when after minification process, migrations cannot be loaded because their name is changed to some random value, this happens mostly when using react-native/expo.

There are several topics about this issue in facebook/metro and typeorm repositories, but I couldn't find any pull request for it. It could be fixed by changing the default minifier, but that can break other stuff.

There are also a few steps before merging this PR:

- [x] If this field sould be called 'migrationName'? Other options could be 'displayName' or simply 'name'.

- [x] Should migrations generated with the CLI contain this attribute by default?

**EDIT**

Created a babel-plugin to automatically add name property:

```js
module.exports = function({ types: t }, options) {
  return {
    visitor: {
      ClassDeclaration(path) {
        const migrationName = t.stringLiteral(path.node.id.name)
        const migrationTimestamp = parseInt(migrationName.value.substr(-13), 10)

        if (!isNaN(migrationTimestamp)) {
          const prop = t.classProperty(t.identifier(options.property), migrationName, null, null)

          path.node.body.body.push(prop)
        }
      },
    },
  }
}
```

```js
{
  presets: ['module:metro-react-native-babel-preset'],
  plugins: [
    ['babel-plugin-typeorm-migration-name', { property: 'migrationName' }]
  ],
}
```

pleerock commented:
> If this field sould be called 'migrationName'? Other options could be 'displayName' or simply 'name'. 

simply name looks fine

> Should migrations generated with the CLI contain this attribute by default?

not sure, but we can add it

leonardofalk commented:
> > If this field sould be called 'migrationName'? Other options could be 'displayName' or simply 'name'.
> 
> simply name looks fine
> 
> > Should migrations generated with the CLI contain this attribute by default?
> 
> not sure, but we can add it

Done, changed to name and added it to the migration template.
